### PR TITLE
Replace setup.py with pyproject.toml

### PR DESCRIPTION
**Pull request by alifbe**
Solves #419

### CODE REVIEW

File: .github/workflows/pyscal.yml [hunk 1]
```diff
@@ -59,7 +59,7 @@ jobs:
       - name: Lint with black
         if: ${{ always() }}
         run: |
-          black --check --force-exclude="pyscal/version.py" pyscal/*py tests/test_*py setup.py docs/conf.py
+          black --check --force-exclude="pyscal/version.py" pyscal/*py tests/test_*py docs/conf.py
```
Comment (mferrera) [inline .github/workflows/pyscal.yml:62]: I'm skeptical that globbing only py files is needed here, but not really relevant to the purpose of this pr :)

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+requires = [
+    "setuptools>=45.0",
+    "setuptools_scm[toml]>=6.2",
+    "wheel>=0.38.1",
+]
+
+build-backend = "setuptools.build_meta"
+
+[tool.setuptools]
+packages = ["pyscal"]
```
Comment (mferrera) [inline pyproject.toml:12]: Is this needed because this repository isn't `src/` based?

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,117 @@
+[build-system]
+
+requires = [
+    "setuptools>=45.0",
+    "setuptools_scm[toml]>=6.2",
+    "wheel>=0.38.1",
```
Comment (mferrera) [inline pyproject.toml:6]: I don't think you need to include wheel

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+"mypy",
+"pre-commit",
+"pytest",
+"pytest-cov",
+"pytest-mock",
+"rstcheck",
+"sphinx<7",
+"sphinx-argparse",
+"sphinx-autodoc-typehints",
+"sphinx_rtd_theme",
```
Comment (mferrera) [inline pyproject.toml:68]: These lack an indent, and

```text
"autoapi",
"sphinx<7",
"sphinx-argparse",
"sphinx-autodoc-typehints",
"sphinx_rtd_theme",
```
should only be needed in the `docs` optionals.

I think you can remove `pre-commit` now too, and you might consider adding `pytest-xdist` 😄

File: pyproject.toml [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    "sphinx<7",
+    "sphinx-argparse",
+    "sphinx-autodoc-typehints",
+    "sphinx_rtd_theme",
+]
+
+[project.urls]
+homepage = "https://github.com/equinor/pyscal"
+documentation = "https://equinor.github.io/pyscal/"
+repository = "https://github.com/equinor/pyscal"
```
Comment (mferrera) [inline pyproject.toml:82]: Consider capitalizing `homepage` et al -- these show up as links on pypi

File: pyproject.toml (hunk 4 above)
Comment (mferrera) [reply pyproject.toml:68]: Note that if you move the docs dependencies out of the `tests` section, you'll need to update ci to install as `pip install .[tests,docs]`, not just `pip install .[tests]`

https://github.com/equinor/pyscal/blob/c8c385b0250f6ba340b18fe96bc0d4a44cf8fea4/.github/workflows/pyscal.yml#L53

File: pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+# that level in subscript
+disable = "missing-module-docstring"
+
+[tool.pytest.ini_options]
+norecursedirs = [
+    ".git",
+]
+xfail_strict = "True"
+
+addopts = "-ra --strict-markers --ignore=docs/conf.py --ignore=setup.py --ignore=.eggs"
```
Comment (mferrera) [inline pyproject.toml:108]: You can remove `--ignore=setup.py` now

File: pyproject.toml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+[tool.black]
+line-length = 88
+
+[tool.isort]
+profile = "black"
+skip = "pyscal/__init__.py"
+
+[tool.mypy]
+ignore_missing_imports = true
```
Comment (mferrera) [inline pyproject.toml:95]: Maybe this is broader than is needed? mypy.ini had more specific constraints, but I have no tested whether or not this makes a difference

File: ci/testkomodo.sh [hunk 8]
```diff
@@ -1,5 +1,5 @@
 install_test_dependencies () {
-  pip install -r test_requirements.txt
+  pip install ".[tests,docs]"
 }
```
Comment (mferrera) [inline ci/testkomodo.sh:3]: I don't think you'll need to includes `docs` here

**Review by mferrera — APPROVED** [review]
Sorry for the delay! I'm approving so you can merge as needed but I think there are some things you could address if you wish

File: .github/workflows/pyscal.yml (hunk 1 above)
Comment (alifbe) [reply .github/workflows/pyscal.yml:62]: that's true. I have removed the globbing part.

File: pyproject.toml (hunk 2 above)
Comment (alifbe) [reply pyproject.toml:12]: yes. I was tempted to move `pyscal` folder inside `src` But decided to use this instead. Any other benefit of using `src/` based?

File: pyproject.toml (hunk 4 above)
Comment (alifbe) [reply pyproject.toml:68]: thanks. I saw your changes in `subscript` Now it's a good time to do the changes with `pyscal` and it cuts the test time by more than 50% :+1:

File: pyproject.toml (hunk 5 above)
Comment (alifbe) [reply pyproject.toml:82]: thanks for the tips :+1:

File: pyproject.toml (hunk 3 above)
Comment (alifbe) [reply pyproject.toml]: You are right. I have excluded `wheel` now

File: pyproject.toml (hunk 7 above)
Comment (alifbe) [reply pyproject.toml]: Agree. I constrained it specific modules now.

````
[[tool.mypy.overrides]]
module = ["numpy.*", "pyscal.version", "pandas.*", "matplotlib.*", "scipy.*", "xlrd.*", "openpyxl.*"
]
ignore_missing_imports = true
````

File: pyproject.toml (hunk 2 above)
Comment (alifbe) [reply pyproject.toml:11]: Good point. I'll adopt this on next PR :+1:

**Pull request merged**